bin/brew

```diff
@@ -44,4 +44,5 @@ fi
 
 HOMEBREW_LIBRARY="$HOMEBREW_REPOSITORY/Library"
 
-source "$HOMEBREW_LIBRARY/Homebrew/brew.sh"
+set > /tmp/homebrew-env
+env -i "$HOMEBREW_LIBRARY/Homebrew/utils/shell_env.sh" "$@"
```

Comment:
Nice 👍. I think it's maybe worth making this conditional for now on the presence of the `HOMEBREW_ENV_FILTERING` environment variable. This means we can do a gradual rollout of this feature.